### ISSUE TITLE
Parenthesized expressions break multiple analyzers that walk syntax chains

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#896]
## Summary

Parenthesized expressions around Moq `Setup` calls (e.g., `(mock.Setup(...)).Returns(42)`) break syntax tree walking in multiple analyzers. Issue #887 fixed the most visible case (Moq1203 false positive), but the underlying problem affects other analyzers as false negatives.

## Root cause

Analyzers walk the Roslyn syntax tree in one of two directions:

- **UP** from `Setup(...)` through `.Parent` to find chained calls like `.Returns()`, `.Throws()`
- **DOWN** from `Returns()`/`Callback()` through `.Expression` to find the originating `Setup()` call

When parentheses wrap part of the chain, a `ParenthesizedExpressionSyntax` node sits between the expected `InvocationExpressionSyntax` and `MemberAccessExpressionSyntax` nodes. Pattern matches like `memberAccess.Expression is InvocationExpressionSyntax` fail because the expression is actually `ParenthesizedExpressionSyntax`.

## Affected analyzers

| Analyzer | ID | Walk direction | Effect | Symptom |
|----------|-----|---------------|--------|---------|
| MethodSetupShouldSpecifyReturnValue | Moq1203 | UP from Setup | Can't find Returns/Throws | **False positive** (fixed in #887) |
| ReturnsAsyncShouldBeUsedForAsync | Moq1206 | DOWN from Returns | Can't find Setup | **False negative** (misses diagnostic) |
| CallbackSignatureShouldMatch | Moq1100 | DOWN from Callback | Can't find Setup | **False negative** (misses diagnostic) |

### Moq1206: ReturnsAsyncShouldBeUsedForAsyncMethodsAnalyzer

`FindSetupInvocation` at `ReturnsAsyncShouldBeUsedForAsyncMethodsAnalyzer.cs:101-104`:

```csharp
if (returnsInvocation.Expression is MemberAccessExpressionSyntax memberAccess &&
    memberAccess.Expression is InvocationExpressionSyntax setupInvocation && // <-- fails here
    ...
```

For `(mock.Setup(x => x.GetAsync())).Returns(async () => "value")`:
- `returnsInvocation.Expression` = `(mock.Setup(...)).Returns` (MemberAccessExpressionSyntax, OK)
- `memberAccess.Expression` = `(mock.Setup(...))` (ParenthesizedExpressionSyntax, NOT InvocationExpressionSyntax)
- Returns null, diagnostic is silently skipped

### Moq1100: CallbackSignatureShouldMatchMockedMethodAnalyzer

`FindSetupMethodFromCallbackInvocation` at `SemanticModelExtensions.cs:18-19`:

```csharp
InvocationExpressionSyntax? invocation = expression as InvocationExpressionSyntax;
if (invocation?.Expression is not MemberAccessExpressionSyntax method)
    return null;
```

For `(mock.Setup(x => x.Foo(It.IsAny<string>()))).Callback((int x) => { })`:
- First iteration: Callback invocation found, not Setup, walks deeper
- `expression = method.Expression` = `(mock.Setup(...))` (ParenthesizedExpressionSyntax)
- Second iteration: `expression as InvocationExpressionSyntax` = null
- Returns null, diagnostic is silently skipped

## Why only Moq1203 was reported

False positives are loud: red squiggles appear on valid code. Users notice immediately.

False negatives are silent: the analyzer just doesn't flag code it should. Nobody notices the missing warning.

## Suggested fix

Create a shared helper (similar to `GetParentSkippingParentheses` added in #887) that unwraps `ParenthesizedExpressionSyntax` when walking DOWN the tree:

```csharp
internal static ExpressionSyntax UnwrapParentheses(this ExpressionSyntax expression)
{
    while (expression is ParenthesizedExpressionSyntax parens)
    {
        expression = parens.Expression;
    }
    return expression;
}
```

Apply it in `FindSetupInvocation`, `FindSetupMethodFromCallbackInvocation`, and any other downward chain walker.

## Related

- #887 (Moq1203 false positive, fixed)
